### ISSUE TITLE
Custom Material with AlphaMode::Blend cause flicker

### ISSUE THREAD

**Issue by stringslength** [bevyengine/bevy#8612]
## Bevy version

bevy = "0.10.1"

## \[Optional\] Relevant system information

> the Rust version you're using (you can get this by running `cargo --version`)

cargo 1.68.2 (6feb7c9cf 2023-03-26)

> the operating system or browser used, including its version

Windows 10, Desktop App

> If your bug is rendering-related, copy the adapter info that appears when you run Bevy.

```ignore
AdapterInfo { name: "NVIDIA GeForce GTX 1070", vendor: 4318, device: 7041, device_type: DiscreteGpu, driver: "NVIDIA", driver_info: "516.94", backend: Vulkan }
```

## What you did

I build example https://bevyengine.org/examples/shader/shader-material/.
Then I add the green-colored outer-cube to example, like this (ADD START - ADD END) :

```
/// set up a simple 3D scene
fn setup(
    mut commands: Commands,
    mut meshes: ResMut<Assets<Mesh>>,
    mut materials: ResMut<Assets<CustomMaterial>>,
    asset_server: Res<AssetServer>,
) {
    // cube
    commands.spawn(MaterialMeshBundle {
        mesh: meshes.add(Mesh::from(shape::Cube { size: 1.0 })),
        transform: Transform::from_xyz(0.0, 0.5, 0.0),
        material: materials.add(CustomMaterial {
            color: Color::BLUE,
            color_texture: Some(asset_server.load("branding/icon.png")),
            alpha_mode: AlphaMode::Blend,
        }),
        ..default()
    });

    // ADD START
    commands.spawn(MaterialMeshBundle {
        mesh: meshes.add(Mesh::from(shape::Cube { size: 3.0 })),
        transform: Transform::from_xyz(0.0, 0.5, 0.0),
        material: materials.add(CustomMaterial {
            color: Color::GREEN,
            color_texture: Some(asset_server.load("branding/icon.png")),
            alpha_mode: AlphaMode::Blend,
        }),
        ..default()
    });
    // ADD END

    // camera
    commands.spawn(Camera3dBundle {
        transform: Transform::from_xyz(-2.0, 2.5, 5.0).looking_at(Vec3::ZERO, Vec3::Y),
        ..default()
    });
}
```

## What went wrong

My expect : show green-cube because blue-cube is hided from it.
Result : blue-cube is displayed flickering. I attach a movie.

https://github.com/bevyengine/bevy/assets/51050009/ec3c21b1-a1e6-4446-bb93-1ca460cb6dea

**Comment by stringslength** [comment]
I suppose that this issue is related to transparent rendering limitation: #2223

**Comment by nicopap** [comment]
You can use `AlphaMode::Mask` to avoid the flickering. It also handles shadows properly. `Blend` is really only good when the material is generally semi-transparent, not a "all opaque/not opaque" style sprite.

I'll close this issue as duplicate, since it can't be fixed without fixing #2223 (and fixing 2223 would fix this)
